src/components/Sidebars/Generate.js

```diff
@@ -1,84 +1,31 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
 import { MDAI } from '@makerdao/dai-plugin-mcd';
 import { Text, Input, Grid, Button } from '@makerdao/ui-components-core';
 import Info from './shared/Info';
 import InfoContainer from './shared/InfoContainer';
-import useStore from 'hooks/useStore';
-import useValidatedInput from 'hooks/useValidatedInput';
+import useCdp from 'hooks/useCdp';
 import useLanguage from 'hooks/useLanguage';
-import { getCdp, getDebtAmount, getCollateralAmount } from 'reducers/cdps';
-import { calcCDPParams } from '../../utils/cdp';
-import { add, subtract, greaterThan } from '../../utils/bignumber';
-import {
-  formatCollateralizationRatio,
-  formatLiquidationPrice,
-  safeToFixed
-} from '../../utils/ui';
 import useMaker from '../../hooks/useMaker';
-import RatioDisplay, { RatioDisplayTypes } from 'components/RatioDisplay';
 
 const Generate = ({ cdpId, reset }) => {
   const { lang } = useLanguage();
   const { maker, newTxListener } = useMaker();
-  const [daiAvailable, setDaiAvailable] = useState(0);
-  const [liquidationPrice, setLiquidationPrice] = useState(0);
-  const [collateralizationRatio, setCollateralizationRatio] = useState(0);
+  const [amount, setAmount] = useState('');
 
-  const [storeState] = useStore();
-  const cdp = getCdp(cdpId, storeState);
-  const { liquidationRatio } = cdp;
+  const cdp = useCdp(cdpId);
 
-  const collateralAmount = getCollateralAmount(cdp, false);
-  const debtAmount = getDebtAmount(cdp, false);
-  const undercollateralized = greaterThan(
-    liquidationRatio,
-    collateralizationRatio
-  );
-
-  const dustLimit = cdp.dust ? cdp.dust : 0;
-
-  // minFloat uses <= so it won't work for dustLimit
-  const dustLimitValidation = value =>
-    greaterThan(dustLimit, add(value, debtAmount));
-
-  const [amount, , onAmountChange, amountErrors] = useValidatedInput(
-    '',
-    {
-      maxFloat: daiAvailable,
-      isFloat: true,
-      custom: {
-        dustLimit: dustLimitValidation
-      }
-    },
-    {
-      maxFloat: () => lang.action_sidebar.cdp_below_threshold,
-      dustLimit: () =>
-        lang.formatString(lang.cdp_create.below_dust_limit, dustLimit)
-    }
-  );
-
-  useEffect(() => {
-    const amountToGenerate = amount || 0;
-    const {
-      liquidationPrice,
-      collateralizationRatio,
-      daiAvailable
-    } = calcCDPParams({
-      ilkData: cdp,
-      gemsToLock: collateralAmount,
-      daiToDraw: add(debtAmount, amountToGenerate)
-    });
+  const dustLimit = cdp.dust || 0;
+  const amountToGenerate = amount || 0;
```

Comment:
Also going back on the discussion with BigNumbers, could these be `BigNumber(0)` instead?

Reply:
imo if we have `cdp.dust` coming in as a `bignumber` (which would be a reasonable change), then `BigNumber(0)` would make sense there. in the `amount` case, i think the raw value from input's `onChange` is one of the few places where it's ok not to use big numbers. we could at some point create a big number input component similar to [what 0x uses](https://github.com/0xProject/0x-launch-kit-frontend/blob/662daf332e9f09b498c17fa50ed84b9c37e1c990/src/components/common/big_number_input.tsx#L93) that uses the raw value internally, but casts it as a big number for the onChange function passed in as a prop.

Reply:
Ya, that's a great idea. I'd be on board with that

Reply:
If it's meant to represent the input value, it would be appropriate for it to be a string

Reply:
agreed